### PR TITLE
Remove roscore plugin.

### PR DESCRIPTION
**Pull request by kyrofa**
This PR resolves LP: [#1539169](https://bugs.launchpad.net/snapcraft/+bug/1539169) by removing the roscore plugin and replacing it with an `include-roscore` option on the Catkin plugin.

### CODE REVIEW

File: docs/ros-snap.md [hunk 1]
```diff
@@ -345,18 +353,20 @@ You can transfer your newly-minted .snap to your Ubuntu Core machine and install
 it at the same time via `snappy-remote`, for example:
 
     $ snappy-remote --url=ssh://<host>:<port> install \
-      ros-talker-and-listener_1.0_amd64.snap
+      ros-example_1.0_amd64.snap
 
-Now on the Ubuntu Core machine, take a look in `/apps/bin/`, and you'll see the
-binary you requested, called `ros-talker-and-listener.launch-project`. Test it
+Now on the Ubuntu Core machine, take a look in `/saps/bin/`, and you'll see the
```
Comment (sergiusens) [inline docs/ros-snap.md]: s|saps/bin|snaps/bin|

File: snapcraft/plugins/catkin.py [hunk 2]
```diff
@@ -166,11 +178,25 @@ def pull(self):
...
+                                                        rosdep)
+
+        # If the package requires roscore, resolve it into a system dependency
+        # as well.
+        if self.options.include_roscore:
+            roscore_dependency = rosdep.resolve_dependency('ros_core')
+            if roscore_dependency:
+                system_dependencies.add(roscore_dependency)
+            else:
+                raise RuntimeError('Unable to determine system dependency for '
```
Comment (sergiusens) [inline snapcraft/plugins/catkin.py]: maybe just put the entire string on the line below

File: docs/ros-snap.md (hunk 1 above)
Comment (kyrofa) [inline docs/ros-snap.md]: Eww, thanks.

File: snapcraft/plugins/catkin.py (hunk 2 above)
Comment (kyrofa) [inline snapcraft/plugins/catkin.py]: Done.

File: docs/ros-snap.md (hunk 1 above)
Comment (kyrofa) [inline docs/ros-snap.md]: Done.

**Comment by sergiusens** [comment]
Nice work

**Pull request merged**